File: Leetcode/go/3541.go
package main

func maxFreqSum(s string) int {
	var lowerAlphabet [26]int

	for _, c := range s {
		lowerAlphabet[c-'a']++
	}

	maxConsonant, maxVowel := 0, 0
	for i, count := range lowerAlphabet {
		c := 'a' + byte(i)
		if c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' {
			if count > maxVowel {
				maxVowel = count
			}
		} else {
			if count > maxConsonant {
				maxConsonant = count
			}
		}
	}

	return maxConsonant + maxVowel
}

File: Leetcode/python/3541.py
class Solution:
    def maxFreqSum(self, s: str) -> int:
        lowerAlphabet = [0] * 26

        for i in s:
            lowerAlphabet[ord(i) - ord('a')] += 1
        
        maxConsonant, maxVowel = 0, 0
        vowels = "aeiou"
        for i in range(len(lowerAlphabet)):
            if chr(i + ord('a')) in vowels:
                maxVowel = max(maxVowel, lowerAlphabet[i])
            else:
                maxConsonant = max(maxConsonant, lowerAlphabet[i])
        
        return maxConsonant + maxVowel